File: sdk/src/main/java/com/cxense/cxensesdk/model/UserDataRequest.java
package com.cxense.cxensesdk.model;


import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

/**
 * Request object for getting user from server
 *
 * @author Dmitriy Konopelkin (dmitry.konopelkin@cxense.com) on (2017-06-05).
 */
@SuppressWarnings({"UnusedDeclaration", "WeakerAccess"}) // Public API.
public final class UserDataRequest extends UserIdentity {
    @SerializedName("groups")
    List<String> groups;
    @SerializedName("recent")
    Boolean isRecent;
    @SerializedName("identityTypes")
    List<String> identityTypes;

    public UserDataRequest(@NonNull UserIdentity identity, @Nullable Collection<String> groups, @Nullable Boolean recent,
                           @Nullable Collection<String> identityTypes) {
        super(identity);
        isRecent = recent;
        if (groups != null)
            this.groups = new ArrayList<>(groups);
        if (identityTypes != null)
            this.identityTypes = new ArrayList<>(identityTypes);
    }
}

File: sdk/src/main/java/com/cxense/cxensesdk/model/UserDataRequest.kt
package com.cxense.cxensesdk.model

import com.google.gson.annotations.SerializedName

/**
 * Request object for getting user from server
 *
 */
class UserDataRequest(
    userIdentity: UserIdentity,
    @SerializedName("groups") val groups: List<String>? = null,
    @SerializedName("recent") val recent: Boolean? = null,
    @SerializedName("identityTypes") val identityTypes: List<String>? = null
) : UserIdentity(userIdentity.type, userIdentity.id)
